### DISCUSSION TITLE
Restoring soft-deleted structure entry should put entry back at same level in structure

### DISCUSSION THREAD

**Discussion by RitterKnightCreative** [craftcms/cms#8630] [Ideas]
### Description
Love the soft delete feature in Craft 3.1.   Client removed an entry today (of course by accident), we were able to easily restore it back but noticed since the item was in a structure section, the restored entry got placed at the _bottom_ of the structure section instead of where it used to be.

While that's not terrible, it'd be great if it saved its location in the hierarchy... in this case, it was a page that doubled as a menu item.

### Steps to reproduce

1. Remove entry.
2. Restore it.

### Additional info

- Craft version: 3.1

**Comment by brandonkelly** [comment]
Yep, i spent more time than I’d like to admit trying to get that working. It’s a complicated problem unfortunately, but will leave this open because I do intend to revisit this at some point when I have more time.

Restored entries should at least (generally) be re-inserted under their old parent entries, so that’s something.
